node/src/components/linear_chain/state.rs

```diff
@@ -299,6 +351,12 @@ impl LinearChain {
             // noop.
             self.cache_signatures(*known_signatures.clone());
         }
+        if let Some(key_block_info) = self.key_block_info.get(&fs.era_id) {
+            let is_bonded = key_block_info
+                .validator_weights()
+                .contains_key(&fs.public_key);
+            return self.handle_is_bonded(signatures, fs, is_bonded);
+        }
```

Comment:
Was the lack of this code a bug? What scenario does it handle which wasn't handle before?

Reply:
No, it just avoids the roundtrip of asking the contract runtime for the set of validators in case we already know it from the key block info.